src/gc-debug.c

```diff
@@ -626,8 +627,33 @@ void gc_debug_print_status(void)
 #ifdef OBJPROFILE
 static htable_t obj_counts[3];
 static htable_t obj_sizes[3];
+static int grab_objprofile = 0;
+JL_STREAM* objprofile_output = NULL;
+char objprofile_name[100];
+
+/* JL_DLLEXPORT return obj_counts, obj_sizes in some julia array or somesuch */
+JL_DLLEXPORT void jl_grab_objprofile(char* name)
+{
+    if (name == NULL) {
+        objprofile_name[0] = 0;
+    } else {
+        memset(objprofile_name, '\0', sizeof(objprofile_name));
+        strncpy(objprofile_name, name, 80);
+    }
+    jl_safe_printf("jl_grab_objprofile triggered with name %s", name);
+
+    grab_objprofile = 1;
+    // jl_gc_collect(JL_GC_FULL);
+}
+
+JL_DLLEXPORT void jl_set_objprofile_output_file(void* stream)
+{
+    objprofile_output = (JL_STREAM*)stream;
+}
+
 void objprofile_count(void *ty, int old, int sz)
 {
+    if (!grab_objprofile) return;
```

Comment:
We should see if there is any performance benefit to inlining this check into callers.